tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -227,7 +227,9 @@ impl<S: Subscriber> Layer<S> for EnvFilter {
             .with(|scope| {
                 for filter in scope.iter() {
                     if filter >= level {
-                        return true;
+                        if self.dynamics.match_target(metadata) {
```

Comment:
I don't think this is implementing the correct behavior. It looks like this will return `true` only if _all_ the dynamic directives in the filter match the metadata's target?